src/EnergyPlus/api/datatransfer.cc

```diff
@@ -279,16 +334,28 @@ int getInternalVariableHandle(const char* type, const char* key) {
 }
 
 Real64 getInternalVariableValue(int handle) {
-    if (handle == 0) {
-        return 1;
+    if (handle >= 1 && handle <= (int)EnergyPlus::DataRuntimeLanguage::EMSInternalVarsAvailable.size()) {
```

Comment:
At line 402 below you use size-1 (in the if test), here just size. I didn't check if these arrays are 1-based index verses 0-based index, just pointing this out.

Reply:
This does look OK. handle vs timeIndex at line 402.